lib/galaxy/jobs/dynamic_tool_destination.py

```diff
@@ -1446,11 +1598,21 @@ def map_tool_to_destination(
         parser.print_help()
         sys.exit(1)
 
-    if args.check_config:
-        valid_config = parse_yaml(path=args.check_config, return_bool=True)
+    if args.check_config and args.job_config:
+        valid_config = parse_yaml(path=args.check_config,
+                                  job_conf_path=args.job_config, return_bool=True)
+
+    elif args.check_config:
+        valid_config = parse_yaml(path=args.check_config,
+                                  job_conf_path="/config/job_conf.xml", return_bool=True)
+
+    elif args.job_config:
+        valid_config = parse_yaml(path="/config/tool_destinations.yml",
+                                  job_conf_path=args.job_config, return_bool=True)
 
     else:
-        valid_config = parse_yaml(path="/config/tool_destinations.yml", return_bool=True)
+        valid_config = parse_yaml(path="/config/tool_destinations.yml",
```

Comment:
also check for other version of the job_conf.xml.sample_advanced and/or job_conf.xml.sample_basic